### ISSUE TITLE
Implement support of multiple sources in relation

### ISSUE THREAD

**Issue by Nayjest** [Nayjest/di-hub#3]
Example:
```php
new ComplexRelationDefinition(
    'target_id', 
    ['source1_id', 'source2_id'],
    function(&$taget, $source1, $source2) {
        &$target = $source1 + $source2;
    }
);
```

**Comment by Nayjest** [comment]
Looks like example code equivalent to:
```php
$hub->addDefinitions([
    new ItemDefinition('tmp', [null, null]),
    new RelationDefinition('tmp', 'source1_id', function(&$t, $s) {
        $t[0] = $s;
    }),
    new RelationDefinition('tmp', 'source2_id', function(&$t, $s) {
        $t[1] = $s2;
    }),
    new RelationDefinition('target_id', 'tmp', function(&$target, $tmp) {
        $t = $tmp[0] + $tmp[1];
    }),
]);
```

\+ possibility to create private item definitions looks good here
```php
$hub->addDefinition(
    new ItemDefinition('tmp', [null, null], ItemDefinitionm::FLAG_PRIVATE)
);
assert($hub->has('tmp') === false);
```

**Comment by Nayjest** [comment]
Implemented in v0.5-dev
